src/components/elements/table/hooks/useTableSelection.tsx

```diff
@@ -63,6 +64,13 @@ export function useTableSelection<T extends { id: string }>({
   // Clear selection when data changes
   useEffect(() => setSelectedState([]), [rows]);
 
+  // Call onChangeSelected callback if selection changes.
```

Comment:
While refactoring, I thought this could be removed, since we now call `onChangeSelected` elsewhere in the hook whenever selection changes. However, it doesn't quite work correctly for the `useEffect` above on line 65, so I'm re-adding this effect. Behavior should be restored to what it was before.

I also investigated an alternative to avoid these effects, such as returning a `clearSelection` callback from this hook. However, the state ownership is a little circuitous here -- it's `GenericTableWithData` and not `GenericTable` that would need to call `clearSelection` when `rows` changes -- so I think the best solution for now is to keep the same `useEffect`s from before.

Reply:
Does this mean onChangeSelected will be called twice for most changes?


> it doesn't quite work correctly for the useEffect above on line 65, so I'm re-adding this effect

If this effect only needs to be triggered when the effect on line 65 runs, can we just move the contents into the effect on line 65 ?

```
  // Clear selection when data changes
  useEffect(() => { setSelectedState([]);  onChangeSelected?.([])}, [rows]);
```
(or injecting some logic for isSelectControlled if needed, I'm not sure if it would be)

Reply:
(I didn't test this, so I'm sure there's something I'm missing. Just wanted to offer the idea).

Reply:
Yes, this was my first thought too -- that we should just call `onChangeSelected([])` inside of the previous `useEffect`. Thanks for pointing me back to it, I think I see a fix for the bug I previously encountered... 🐞

Reply:
ok, great!